src/core/authentication/authentication.service.ts

```diff
@@ -136,31 +136,34 @@ export class AuthenticationService {
     return agentInfo;
   }
 
-  public async extendSession(sessionToBeExtended: Session): Promise<void> {
+  public async extendSession(sessionToBeExtended: Session): Promise<Session> {
     const bearerToken = await getBearerToken(
       this.kratosPublicUrlServer,
       this.adminPasswordIdentifier,
       this.adminPassword
     );
 
-    await this.tryExtendSession(sessionToBeExtended, bearerToken);
+    return this.tryExtendSession(sessionToBeExtended, bearerToken);
```

Comment:
I've put this await here last time as errors weren't propagated / logged on the main thread - please check whether throwing a kratos error will be logged correctly if you don't await.

Reply:
`extendSession` has a `catch` where it's used